app/src/pages/playground/InvocationParametersForm.tsx

```diff
@@ -16,113 +30,221 @@ const FormField = ({
       );
-    case "maxTokens":
-      return (
-        <TextField
-          label="Max Tokens"
-          value={value?.toString() || ""}
-          type="number"
-          onChange={(value) => onChange(Number(value))}
-        />
-      );
-    case "stop":
+    case "StringListInvocationParameter":
       if (!Array.isArray(value) && value !== undefined) return null;
       return (
         <TextField
-          label="Stop"
+          label={field.label}
+          isRequired={field.required}
           defaultValue={value?.join(", ") || ""}
           onChange={(value) => onChange(value.split(/, */g))}
         />
       );
-    case "seed":
+    case "StringInvocationParameter":
       return (
         <TextField
-          label="Seed"
+          label={field.label}
+          isRequired={field.required}
           value={value?.toString() || ""}
-          type="number"
-          onChange={(value) => onChange(Number(value))}
+          type="text"
+          onChange={(value) => onChange(value)}
         />
       );
+    case "BooleanInvocationParameter":
+      // TODO: add checkbox
+      return null;
     default:
       return null;
   }
 };
 
-export type InvocationParametersChangeHandler = <
-  T extends keyof ModelConfig["invocationParameters"],
->(
-  parameter: T,
-  value: ModelConfig["invocationParameters"][T]
-) => void;
+const toCamelCase = (str: string) =>
+  str.replace(/_([a-z])/g, (_, char) => char.toUpperCase());
+
+const getInvocationParameterValue = (
+  field: InvocationParameter,
+  parameterInput: InvocationParameterInput
+): string | number | string[] | boolean | null | undefined => {
+  if (field.invocationInputField === undefined) {
+    throw new Error("Invocation input field is required");
+  }
+  return parameterInput[
+    toCamelCase(field.invocationInputField) as keyof InvocationParameterInput
+  ];
+};
+
+const makeInvocationParameterInput = (
+  field: InvocationParameter,
+  value: string | number | string[] | boolean | undefined
+): InvocationParameterInput | null => {
+  if (field.invocationName === undefined) {
+    throw new Error("Invocation name is required");
+  }
+  if (field.invocationInputField === undefined) {
+    throw new Error("Invocation input field is required");
+  }
+  return {
+    invocationName: field.invocationName,
+    canonicalName: field.canonicalName,
+    [toCamelCase(field.invocationInputField)]: value,
+  };
+};
 
 type InvocationParametersFormProps = {
-  model: ModelConfig;
-  onChange: InvocationParametersChangeHandler;
+  instance: PlaygroundInstance;
 };
 
 export const InvocationParametersForm = ({
-  model,
-  onChange,
+  instance,
 }: InvocationParametersFormProps) => {
-  const { invocationParameters, provider, modelName } = model;
-  // Get the schema for the incoming provider and model combination.
-  const schema = getInvocationParametersSchema({
-    modelProvider: provider,
-    modelName: modelName || "default",
-  });
+  const { model } = instance;
```

Comment:
I was having some issues when the instance in the parent changed but children were not being re-rendered due to the way we're pullin gth einstance form the store, if that's not happening here no biggy but yeah some weird caching stuff was happening even though the instances are being mapped (new array) in the store

Reply:
Probably something to do with shallow vs non-shallow zustand selectors. I think I may have found an edge case that replicates the behavior you're describing. Gonna try selecting instance out of store by id